tests/kernel/spinlock/src/spinlock_fairness.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2023 Syntacore. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/******************************************************************************
```

Comment:
Just personal preference, but I would prefer to see far fewer comments delineating source file sections and fewer doxygen comments for functions that simply state the obvious.